compiler/cpp/src/lexer/Lexer.cpp

```diff
@@ -0,0 +1,279 @@
+    }
+    currentContext = SECTION;
+    return constructToken(TOK_ARBITRARY, payloadStr.str());
+}
+
+Token Lexer::consumeSection() {
+    char curChar = (char) reader.getLookahead();
+
+    if (isEOF()) return consumeEOF();
+
+    if (isLookAheadCommentLineIdentifier()) return consumeCommentLineIdentifier();
+    if (isLookAheadCommentBlockOpenIdentifier()) return consumeCommentBlockOpenIdentifier();
+    if (isLookAheadCommentBlockCloseIdentifier()) return consumeCommentBlockCloseIdentifier();
+    if (isLookAheadCommentPayloadIdentifier()) return consumeCommentPayloadIdentifier();
+
+    switch (curChar) {
+        case '|': return consumeOr();
+        case '=': return consumeEquals();
+        case '!': return consumeNotEquals();
+        case '>': return consumeGreaterOrGreaterEqual();
+        case '<': return consumeLessOrLessEqual();
+        case '.': return consumeDot();
+        case '{': return consumeBraceOpen();
+        case '}': return consumeBraceClose();
+        case '[': return consumeIndex();
+        case '"': return consumeString();
+    }
+
+    if (isalpha(curChar)) return consumeIdentifierOrKeyword();
+    if (isdigit(curChar)) return consumeNumber();
+
+    throw UnexpectedCharException(curChar, reader.getLineNum(), reader.getColNum());
+}
+
+Token Lexer::consumeEOF() {
+    return constructToken(TOK_EOF);
+}
+
+Token Lexer::consumeOr() {
+    reader.expect('|');
+    return constructToken(TOK_OR);
+}
+
+Token Lexer::consumeEquals() {
+    reader.expect('=');
+    reader.expect('=');
+    return constructToken(TOK_EQUALS);
+}
+
+Token Lexer::consumeNotEquals() {
+    reader.expect('!');
+    reader.expect('=');
+    return constructToken(TOK_NOT_EQUALS);
+}
+
+Token Lexer::consumeGreaterOrGreaterEqual() {
+    reader.expect('>');
+    if (reader.getLookahead() == '=') {
+        reader.expect('=');
+        return constructToken(TOK_GREATER_EQUAL);
+    }
+    return constructToken(TOK_GREATER);
+}
+
+Token Lexer::consumeLessOrLessEqual() {
+    reader.expect('<');
+    if (reader.getLookahead() == '=') {
+        reader.expect('=');
+        return constructToken(TOK_LESS_EQUAL);
+    }
+    return constructToken(TOK_LESS);
+}
+
+Token Lexer::consumeDot() {
+    reader.expect('.');
+    return constructToken(TOK_DOT);
+}
+
+Token Lexer::consumeBraceOpen() {
+    reader.expect('{');
+    currentContext = PAYLOAD;
+    return constructToken(TOK_BRACE_OPEN);
+}
+
+Token Lexer::consumeBraceClose() {
+    reader.expect('}');
+    if (!isLookAheadCommentBlockCloseIdentifier()) currentContext = ARBITRARY;
+    return constructToken(TOK_BRACE_CLOSE);
+}
+
+Token Lexer::consumeIndex() {
+    reader.expect('[');
+    Token indexToken = consumeNumber();
+    reader.expect(']');
+    return constructToken(TOK_INDEX, indexToken.getValue());
+}
+
+Token Lexer::consumeString() {
+    reader.expect('"');
+    std::stringstream stringStr;
+    while(reader.getLookahead() != '"' && reader.getLookahead() != EOF) {
```

Comment:
Yes, that is nice. In my review for your code, I suggested to define EOF as a constant, available to all files, so that you can use it, similar to this